build-system/common/utils.js

```diff
@@ -45,6 +46,31 @@ async function buildRuntime(opt_compiled = false) {
   }
 }
 
+/**
+ * Extracts and validates the config for the given experiment.
+ * @param {string} experiment
+ * @return {Object|null}
+ */
+function getExperimentConfig(experiment) {
+  const config = experimentsConfig[experiment];
+  const valid =
+    config?.name &&
+    config?.define_experiment_constant &&
+    config?.expiration_date_utc &&
+    new Number(new Date(config.expiration_date_utc)) >= Date.now();
+  return valid ? config : null;
+}
+
+/**
+ * Returns the names of all valid experiments.
+ * @return {!Array<string>}
+ */
+function getValidExperiments() {
```

Comment:
Since you're pulling in the config itself, can you use the keys to get the experiment names?

Reply:
Good idea! Done.